File: Backend/src/main/java/com/sencarmarket/module/abonnement/controller/AbonnementController.java
package com.sencarmarket.module.abonnement.controller;

import com.sencarmarket.module.abonnement.dto.AbonnementResponse;
import com.sencarmarket.module.abonnement.dto.BoostAnnonceResponse;
import com.sencarmarket.module.abonnement.dto.CreateAbonnementRequest;
import com.sencarmarket.module.abonnement.dto.CreateBoostRequest;
import com.sencarmarket.module.abonnement.dto.SouscriptionRequest;
import com.sencarmarket.module.abonnement.dto.UtilisateurAbonnementResponse;
import com.sencarmarket.module.abonnement.service.IAbonnementService;
import com.sencarmarket.module.commun.constants.AppMessages;
import com.sencarmarket.module.commun.dto.PaginatedResponse;
import com.sencarmarket.module.commun.security.AccessControlService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

/**
 * Controleur REST pour la gestion des abonnements
 */
@RestController
@RequestMapping("/api/abonnements")
@RequiredArgsConstructor
@Slf4j
@PreAuthorize("isAuthenticated()")
public class AbonnementController {

    private final IAbonnementService abonnementService;
    private final AccessControlService accessControlService;

    // ==================== GESTION DES PLANS D'ABONNEMENT ====================

    /**
     * Creer un nouveau plan d'abonnement - Admin uniquement
     */
    @PostMapping("/plans")
    @PreAuthorize("hasRole('ADMIN')")
    public ResponseEntity<AbonnementResponse> createPlan(@Valid @RequestBody CreateAbonnementRequest request) {
        log.info("Requete de creation d'un plan d'abonnement: {}", request.getNom());
        AbonnementResponse response = abonnementService.createAbonnement(request);
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }

    /**
     * Mettre a jour un plan d'abonnement - Admin uniquement
     */
    @PutMapping("/plans/{id}")
    @PreAuthorize("hasRole('ADMIN')")
    public ResponseEntity<AbonnementResponse> updatePlan(
            @PathVariable UUID id,
            @Valid @RequestBody CreateAbonnementRequest request) {
        log.info("Requete de mise a jour du plan d'abonnement: {}", id);
        AbonnementResponse response = abonnementService.updateAbonnement(id, request);
        return ResponseEntity.ok(response);
    }

    /**
     * Supprimer (desactiver) un plan d'abonnement - Admin uniquement
     */
    @DeleteMapping("/plans/{id}")
    @PreAuthorize("hasRole('ADMIN')")
    public ResponseEntity<Void> deletePlan(@PathVariable UUID id) {
        log.info("Requete de suppression du plan d'abonnement: {}", id);
        abonnementService.deleteAbonnement(id);
        return ResponseEntity.noContent().build();
    }

    /**
     * Obtenir un plan d'abonnement par ID
     */
    @GetMapping("/plans/{id}")
    public ResponseEntity<AbonnementResponse> getPlanById(@PathVariable UUID id) {
        log.info("Requête de récupération du plan d'abonnement: {}", id);
        AbonnementResponse response = abonnementService.getAbonnementById(id);
        return ResponseEntity.ok(response);
    }

    /**
     * Obtenir tous les plans d'abonnement actifs
     */
    @GetMapping("/plans")
    public ResponseEntity<List<AbonnementResponse>> getAllPlans() {
        log.info("Requête de récupération de tous les plans d'abonnement");
        List<AbonnementResponse> response = abonnementService.getAllAbonnements();
        return ResponseEntity.ok(response);
    }

    // ==================== SOUSCRIPTION ====================

    /**
     * Souscrire à un plan d'abonnement
     */
    @PostMapping("/souscription")
    public ResponseEntity<UtilisateurAbonnementResponse> subscribe(
            @Valid @RequestBody SouscriptionRequest request,
            Authentication authentication) {
        applyAuthenticatedUtilisateurId(request, authentication);
        accessControlService.checkOwnerOrAdmin(authentication, request.getUtilisateurId(), AppMessages.ACCESS_DENIED_RESOURCE);
        log.info("Requête de souscription - Utilisateur: {}, Abonnement: {}", 
                request.getUtilisateurId(), request.getAbonnementId());
        UtilisateurAbonnementResponse response = abonnementService.subscribe(request);
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }

    /**
     * Renouveler un abonnement
     */
    @PostMapping("/utilisateurs/{utilisateurId}/renew")
    public ResponseEntity<UtilisateurAbonnementResponse> renewSubscription(
            @PathVariable UUID utilisateurId,
            Authentication authentication) {
        accessControlService.checkOwnerOrAdmin(authentication, utilisateurId, AppMessages.ACCESS_DENIED_RESOURCE);
        log.info("Requête de renouvellement pour l'utilisateur: {}", utilisateurId);
        UtilisateurAbonnementResponse response = abonnementService.renewSubscription(utilisateurId);
        return ResponseEntity.ok(response);
    }

    /**
     * Annuler un abonnement
     */
    @PostMapping("/utilisateurs/{utilisateurId}/cancel")
    public ResponseEntity<Void> cancelSubscription(@PathVariable UUID utilisateurId, Authentication authentication) {
        accessControlService.checkOwnerOrAdmin(authentication, utilisateurId, AppMessages.ACCESS_DENIED_RESOURCE);
        log.info("Requête d'annulation pour l'utilisateur: {}", utilisateurId);
        abonnementService.cancelSubscription(utilisateurId);
        return ResponseEntity.noContent().build();
    }

    /**
     * Obtenir l'abonnement actif d'un utilisateur
     */
    @GetMapping("/utilisateurs/{utilisateurId}/actif")
    public ResponseEntity<UtilisateurAbonnementResponse> getActiveSubscription(
            @PathVariable UUID utilisateurId,
            Authentication authentication) {
        accessControlService.checkOwnerOrAdmin(authentication, utilisateurId, AppMessages.ACCESS_DENIED_RESOURCE);
        log.info("Requête de récupération de l'abonnement actif pour: {}", utilisateurId);
        UtilisateurAbonnementResponse response = abonnementService.getActiveSubscription(utilisateurId);
        
        if (response == null) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.ok(response);
    }

    /**
     * Obtenir l'historique des abonnements d'un utilisateur
     */
    @GetMapping("/utilisateurs/{utilisateurId}")
    public ResponseEntity<PaginatedResponse<UtilisateurAbonnementResponse>> getSubscriptionsHistory(
            @PathVariable UUID utilisateurId,
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "10") int size,
            Authentication authentication) {
        accessControlService.checkOwnerOrAdmin(authentication, utilisateurId, AppMessages.ACCESS_DENIED_RESOURCE);
        log.info("Requête de récupération de l'historique pour l'utilisateur: {}", utilisateurId);
        PaginatedResponse<UtilisateurAbonnementResponse> response = 
                abonnementService.getSubscriptionsByUtilisateur(utilisateurId, page, size);
        return ResponseEntity.ok(response);
    }

    // ==================== BOOST ====================

    /**
     * Créer un boost pour une annonce
     */
    @PostMapping("/boosts")
    public ResponseEntity<BoostAnnonceResponse> createBoost(@Valid @RequestBody CreateBoostRequest boost) {
        log.info("Requête de création d'un boost pour l'annonce: {}", boost.getAnnonceLocationId());
        BoostAnnonceResponse response = abonnementService.createBoost(boost);
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }

    /**
     * Mettre à jour un boost
     */
    @PutMapping("/boosts/{id}")
    public ResponseEntity<BoostAnnonceResponse> updateBoost(
            @PathVariable UUID id,
            @Valid @RequestBody CreateBoostRequest boost) {
        log.info("Requête de mise à jour du boost: {}", id);
        BoostAnnonceResponse response = abonnementService.updateBoost(id, boost);
        return ResponseEntity.ok(response);
    }

    /**
     * Supprimer un boost
     */
    @DeleteMapping("/boosts/{id}")
    public ResponseEntity<Void> deleteBoost(@PathVariable UUID id) {
        log.info("Requête de suppression du boost: {}", id);
        abonnementService.deleteBoost(id);
        return ResponseEntity.noContent().build();
    }

    /**
     * Obtenir un boost par ID
     */
    @GetMapping("/boosts/{id}")
    public ResponseEntity<BoostAnnonceResponse> getBoostById(@PathVariable UUID id) {
        log.info("Requête de récupération du boost: {}", id);
        BoostAnnonceResponse response = abonnementService.getBoostById(id);
        return ResponseEntity.ok(response);
    }

    /**
     * Obtenir les boosts actifs pour un véhicule
     */
    @GetMapping("/vehicules/{vehiculeId}/boosts")
    public ResponseEntity<List<BoostAnnonceResponse>> getBoostsByVehicule(@PathVariable UUID vehiculeId) {
        log.info("Requête de récupération des boosts pour le véhicule: {}", vehiculeId);
        List<BoostAnnonceResponse> response = abonnementService.getBoostsByVehicule(vehiculeId);
        return ResponseEntity.ok(response);
    }

    private void applyAuthenticatedUtilisateurId(SouscriptionRequest request, Authentication authentication) {
        UUID currentUserId = accessControlService.getCurrentUserId(authentication);
        if (!accessControlService.isAdmin(authentication) || request.getUtilisateurId() == null) {
            request.setUtilisateurId(currentUserId);
        }
    }
}

File: Backend/src/modules/abonnement/abonnement.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  HttpCode,
  HttpStatus,
  Param,
  ParseIntPipe,
  ParseUUIDPipe,
  Post,
  Put,
  Query,
  UseGuards,
} from '@nestjs/common';
import { ApiBearerAuth, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';

import { CurrentUser } from '../../common/decorators/current-user.decorator';
import { PaginatedResponseDto } from '../../common/dto/paginated-response.dto';
import { JwtAuthGuard } from '../../common/guards/jwt-auth.guard';
import type { AuthenticatedUser } from '../../common/types/authenticated-user.type';
import { ApiErrorResponseDto } from '../auth/dto/api-error-response.dto';

import { AbonnementService } from './abonnement.service';
import { AbonnementResponseDto } from './dto/abonnement-response.dto';
import { BoostAnnonceResponseDto } from './dto/boost-annonce-response.dto';
import { CreateAbonnementRequestDto } from './dto/create-abonnement-request.dto';
import { CreateBoostRequestDto } from './dto/create-boost-request.dto';
import { SouscriptionRequestDto } from './dto/souscription-request.dto';
import { UtilisateurAbonnementResponseDto } from './dto/utilisateur-abonnement-response.dto';

@ApiTags('Abonnements')
@UseGuards(JwtAuthGuard)
@ApiBearerAuth()
@Controller('abonnements')
export class AbonnementController {
  constructor(private readonly service: AbonnementService) {}

  @Post('plans')
  @HttpCode(HttpStatus.CREATED)
  @ApiOperation({ summary: 'Créer un plan d\'abonnement' })
  @ApiResponse({ status: 201, type: AbonnementResponseDto, description: 'Plan créé avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  createPlan(
    @Body() request: CreateAbonnementRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<AbonnementResponseDto> {
    return this.service.createPlan(request, user);
  }

  @Put('plans/:id')
  @ApiOperation({ summary: 'Mettre à jour un plan' })
  @ApiResponse({ status: 200, type: AbonnementResponseDto, description: 'Plan mis à jour avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Plan non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  updatePlan(
    @Param('id', new ParseUUIDPipe()) id: string,
    @Body() request: CreateAbonnementRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<AbonnementResponseDto> {
    return this.service.updatePlan(id, request, user);
  }

  @Delete('plans/:id')
  @HttpCode(HttpStatus.NO_CONTENT)
  @ApiOperation({ summary: 'Supprimer un plan (désactivation)' })
  @ApiResponse({ status: 204, description: 'Plan supprimé avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Plan non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  async deletePlan(
    @Param('id', new ParseUUIDPipe()) id: string,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<void> {
    await this.service.deletePlan(id, user);
  }

  @Get('plans/:id')
  @ApiOperation({ summary: 'Obtenir un plan par ID' })
  @ApiResponse({ status: 200, type: AbonnementResponseDto, description: 'Plan récupéré avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Plan non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getPlanById(@Param('id', new ParseUUIDPipe()) id: string): Promise<AbonnementResponseDto> {
    return this.service.getPlanById(id);
  }

  @Get('plans')
  @ApiOperation({ summary: 'Liste des plans actifs' })
  @ApiResponse({ status: 200, description: 'Plans récupérés avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getAllPlans(): Promise<AbonnementResponseDto[]> {
    return this.service.getAllPlans();
  }

  @Post('souscription')
  @HttpCode(HttpStatus.CREATED)
  @ApiOperation({ summary: 'Souscrire à un plan' })
  @ApiResponse({ status: 201, type: UtilisateurAbonnementResponseDto, description: 'Abonnement créé avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Plan non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  subscribe(
    @Body() request: SouscriptionRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<UtilisateurAbonnementResponseDto> {
    return this.service.subscribe(request, user);
  }

  @Post('utilisateurs/:utilisateurId/renew')
  @ApiOperation({ summary: 'Renouveler un abonnement' })
  @ApiResponse({ status: 200, type: UtilisateurAbonnementResponseDto, description: 'Abonnement renouvelé avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Abonnement non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  renewSubscription(
    @Param('utilisateurId', new ParseUUIDPipe()) utilisateurId: string,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<UtilisateurAbonnementResponseDto> {
    return this.service.renewSubscription(utilisateurId, user);
  }

  @Post('utilisateurs/:utilisateurId/cancel')
  @HttpCode(HttpStatus.NO_CONTENT)
  @ApiOperation({ summary: 'Annuler un abonnement' })
  @ApiResponse({ status: 204, description: 'Abonnement annulé avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Abonnement non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  async cancelSubscription(
    @Param('utilisateurId', new ParseUUIDPipe()) utilisateurId: string,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<void> {
    await this.service.cancelSubscription(utilisateurId, user);
  }

  @Get('utilisateurs/:utilisateurId/actif')
  @ApiOperation({ summary: 'Abonnement actif' })
  @ApiResponse({ status: 200, type: UtilisateurAbonnementResponseDto, description: 'Abonnement actif récupéré avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Abonnement non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getActiveSubscription(
    @Param('utilisateurId', new ParseUUIDPipe()) utilisateurId: string,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<UtilisateurAbonnementResponseDto | null> {
    return this.service.getActiveSubscription(utilisateurId, user);
  }

  @Get('utilisateurs/:utilisateurId')
  @ApiOperation({ summary: 'Historique des abonnements' })
  @ApiResponse({ status: 200, description: 'Historique récupéré avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getSubscriptionsHistory(
    @Param('utilisateurId', new ParseUUIDPipe()) utilisateurId: string,
    @Query('page', new ParseIntPipe({ optional: true })) page = 0,
    @Query('size', new ParseIntPipe({ optional: true })) size = 10,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<PaginatedResponseDto<UtilisateurAbonnementResponseDto>> {
    return this.service.getSubscriptionsHistory(utilisateurId, page, size, user);
  }

  @Post('boosts')
  @HttpCode(HttpStatus.CREATED)
  @ApiOperation({ summary: 'Créer un boost' })
  @ApiResponse({ status: 201, type: BoostAnnonceResponseDto, description: 'Boost créé avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  createBoost(
    @Body() request: CreateBoostRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<BoostAnnonceResponseDto> {
    return this.service.createBoost(request, user);
  }

  @Put('boosts/:id')
  @ApiOperation({ summary: 'Mettre à jour un boost' })
  @ApiResponse({ status: 200, type: BoostAnnonceResponseDto, description: 'Boost mis à jour avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Boost non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  updateBoost(
    @Param('id', new ParseUUIDPipe()) id: string,
    @Body() request: CreateBoostRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<BoostAnnonceResponseDto> {
    return this.service.updateBoost(id, request, user);
  }

  @Delete('boosts/:id')
  @HttpCode(HttpStatus.NO_CONTENT)
  @ApiOperation({ summary: 'Supprimer un boost' })
  @ApiResponse({ status: 204, description: 'Boost supprimé avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Boost non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  async deleteBoost(
    @Param('id', new ParseUUIDPipe()) id: string,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<void> {
    await this.service.deleteBoost(id, user);
  }

  @Get('boosts/:id')
  @ApiOperation({ summary: 'Obtenir un boost par ID' })
  @ApiResponse({ status: 200, type: BoostAnnonceResponseDto, description: 'Boost récupéré avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Boost non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getBoostById(@Param('id', new ParseUUIDPipe()) id: string): Promise<BoostAnnonceResponseDto> {
    return this.service.getBoostById(id);
  }

  @Get('vehicules/:vehiculeId/boosts')
  @ApiOperation({ summary: 'Boosts d\'un véhicule' })
  @ApiResponse({ status: 200, description: 'Boosts récupérés avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Véhicule non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getBoostsByVehicule(
    @Param('vehiculeId', new ParseUUIDPipe()) vehiculeId: string,
  ): Promise<BoostAnnonceResponseDto[]> {
    return this.service.getBoostsByVehicule(vehiculeId);
  }
}
